lib/location/scan_wifi.c

```diff
@@ -203,7 +206,7 @@ void scan_wifi_net_mgmt_event_handler(
 int scan_wifi_cancel(void)
 {
 	if (scan_wifi_ready != NULL) {
-		k_sem_reset(scan_wifi_ready);
+		k_sem_give(scan_wifi_ready);
```

Comment:
Is there a log which would show a scenario that this fixes. I can imagine what it fixes but I think we would need to fix the caller side. We are likely doing operations that we shouldn't do after the location request has been cancelled, such as taking the semaphore for waiting for Wi-Fi scan results.